cpp/tests/test_save_results.cpp

```diff
@@ -339,4 +370,129 @@ TEST(TestSaveResult, save_percentiles_and_sums)
+        params.get<mio::osecir::TimeInfectedSevere<double>>()[i]     = 10.;
+        params.get<mio::osecir::TimeInfectedCritical<double>>()[i]   = 8.;
+
+        model.populations[{i, mio::osecir::InfectionState::Exposed}]            = 100;
+        model.populations[{i, mio::osecir::InfectionState::InfectedNoSymptoms}] = 50;
+        model.populations[{i, mio::osecir::InfectionState::InfectedSymptoms}]   = 50;
+        model.populations[{i, mio::osecir::InfectionState::InfectedSevere}]     = 20;
+        model.populations[{i, mio::osecir::InfectionState::InfectedCritical}]   = 10;
+        model.populations[{i, mio::osecir::InfectionState::Recovered}]          = 10;
+        model.populations[{i, mio::osecir::InfectionState::Dead}]               = 0;
+        model.populations.set_difference_from_total({i, mio::osecir::InfectionState::Susceptible}, pop_total);
+
+        params.get<mio::osecir::TransmissionProbabilityOnContact<double>>()[i] = 0.06;
+        params.get<mio::osecir::RelativeTransmissionNoSymptoms<double>>()[i]   = 0.67;
+        params.get<mio::osecir::RecoveredPerInfectedNoSymptoms<double>>()[i]   = 0.09;
+        params.get<mio::osecir::RiskOfInfectionFromSymptomatic<double>>()[i]   = 0.25;
+        params.get<mio::osecir::SeverePerInfectedSymptoms<double>>()[i]        = 0.2;
+        params.get<mio::osecir::CriticalPerSevere<double>>()[i]                = 0.25;
+        params.get<mio::osecir::DeathsPerCritical<double>>()[i]                = 0.3;
+    }
+
+    mio::ContactMatrixGroup& contact_matrix = params.get<mio::osecir::ContactPatterns<double>>();
+    contact_matrix[0] = mio::ContactMatrix(Eigen::MatrixXd::Constant((size_t)nb_groups, (size_t)nb_groups, cont_freq));
+    contact_matrix[0].add_damping(0.7, mio::SimulationTime(30.));
+
+    // setup graph
+    mio::Graph<mio::SimulationNode<mio::osecir::Simulation<>>, mio::MigrationEdge<double>> g;
+    g.add_node(0, model, t0);
+    g.add_node(1, model, t0);
+    g.add_edge(0, 1, Eigen::VectorXd::Constant((size_t)mio::osecir::InfectionState::Count, 0.1));
+    g.add_edge(1, 0, Eigen::VectorXd::Constant((size_t)mio::osecir::InfectionState::Count, 0.1));
+
+    // simulation
+    auto sim = mio::make_migration_sim(t0, dt, std::move(g));
+    sim.advance(tmax);
+
+    // get_results and pair ids
+    auto& results_edge_0                               = sim.get_graph().edges()[0].property.get_migrated();
+    std::vector<mio::TimeSeries<double>> results_edges = {results_edge_0, results_edge_0};
+
+    auto pairs_edges = std::vector<std::pair<int, int>>{};
+    pairs_edges.push_back({0, 1});
+    pairs_edges.push_back({1, 0});
+
+    TempFileRegister file_register;
+    auto results_file_path = file_register.get_unique_path("test_result-%%%%-%%%%.h5");
+    auto save_edges_status = mio::save_edges(results_edges, pairs_edges, results_file_path);
+    ASSERT_TRUE(save_edges_status);
+
+    auto results_from_file = mio::read_result(results_file_path);
```

Comment:
This test does not actually test whether save_edges works. At best, you assert that read_result can deal with what save_edges outputs. But, without making sure this output is correct, that does not mean much.

A better test would be to set up a few (mock) edges with small results as input, and then check whether those are saved as expected - i.e., compare them against a manually written string.
This setup can then also be used to test read_result, by exchanging input and output.

Reply:
adjusted the tests